### PR TITLE
Test predicates pushdown to ClickHouse FLOAT type

### PR DESCRIPTION
**Pull request by ssheikin**
## Additional context and related issues

- https://github.com/trinodb/trino/issues/9998
- https://github.com/trinodb/trino/pull/10000

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseConnectorTest.java [hunk 1]
```diff
@@ -100,10 +100,16 @@ protected QueryRunner createQueryRunner()
...
-        assertUpdate("ALTER TABLE test SET PROPERTIES sample_by = 'p2'");
+        String tableName = "sql_injection_" + randomNameSuffix();
+        try {
+            assertQueryFails("CREATE TABLE " + tableName + " (p1 int NOT NULL, p2 boolean NOT NULL, x VARCHAR) WITH (engine = 'MergeTree', order_by = ARRAY['p1', 'p2'], primary_key = ARRAY['p1', 'p2'], sample_by = 'p2; drop table tpch.nation')", "(?s).*Missing columns: 'p2; drop table tpch.nation.*");
+            assertUpdate("CREATE TABLE " + tableName + " (p1 int NOT NULL, p2 boolean NOT NULL, x VARCHAR) WITH (engine = 'MergeTree', order_by = ARRAY['p1', 'p2'], primary_key = ARRAY['p1', 'p2'], sample_by = 'p2')");
+            assertQueryFails("ALTER TABLE " + tableName + " SET PROPERTIES sample_by = 'p2; drop table tpch.nation'", "(?s).*Missing columns: 'p2; drop table tpch.nation.*");
+            assertUpdate("ALTER TABLE " + tableName + " SET PROPERTIES sample_by = 'p2'");
+        }
+        finally {
+            assertUpdate("DROP TABLE " + tableName);
```
Comment (ebyhr) [inline plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseConnectorTest.java:111]: This assertion may fail if the 2nd `CREATE TABLE` fails for some reason. How about adding `IF EXISTS` option?

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseConnectorTest.java [hunk 2]
```diff
@@ -822,6 +822,74 @@ public void testLargeDefaultDomainCompactionThreshold()
...
+                        c_real_nan real,
+                        c_double double,
+                        c_double_neg_infinity double,
+                        c_double_pos_infinity double,
+                        c_double_nan double)""",
+                List.of("3.14, -infinity(), +infinity(), nan(), 3.14, -infinity(), +infinity(), nan()"))) {
+            assertThat(query("""
+                    SELECT c_real, c_double, c_double_neg_infinity, c_double_pos_infinity, c_double_nan FROM %s
+                    WHERE c_real = real '3.14'""".formatted(table.getName())))
+                    .matches("VALUES (real '3.14', double '3.14', -infinity(), +infinity(), nan())")
```
Comment (ebyhr) [inline plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseConnectorTest.java:845]: Why do we want to check values of `c_real, c_double, c_double_neg_infinity, c_double_pos_infinity, c_double_nan`? That looks unrelated for verifying predicate pushdown. How about adding another column (e.g.`id int`)?

**Review by ebyhr — APPROVED** [review]

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseConnectorTest.java (hunk 2 above)
Comment (ssheikin) [reply plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseConnectorTest.java:845]: I'll just reuse the existing one `c_real`
